### DISCUSSION TITLE
custom classes from a global.css not found

### DISCUSSION THREAD

**Discussion by denu5** [ben-rogerson/twin.macro#525] [Q&A] [answered]
say we import in app.tsx ` import '../styles/global.css';`

```
@layer base {

    /* inspiration https://tailwindcss-typography.netlify.app/ */
    .prose {
        h1, .h1 {
            @apply font-serif font-bold;           
```

should i expect this to work? currently i get **✕ prose was not found**

i assume it's going into this direction: https://github.com/ben-rogerson/twin.macro/discussions/338

**Comment by ben-rogerson** [comment, marked as answer]
Instead of using @layer, add the prose class in a global provider from your css-in-js library and apply `className="prose"` on your container.

**Reply by denu5** [reply]
okey, only to clarify => tw will never pick up those classes right?.. edit: which would make sense, as they don't exist in the tailwind.config

**Reply by ben-rogerson** [reply]
yeah thats right, twin won't pick up any classes defined in a global provider from your css-in-js library.

**Reply by denu5** [reply]
hi ben, thanks for your invididual support :) i hope a final question:

can the opposite work? meaning tailwind classes for regular className props?

did some experiments in your examples:

![image](https://user-images.githubusercontent.com/5692193/131637190-0240c280-8e67-4816-80cb-94c853454bd8.png)

```
    <div tw="bg-electric text-white p-4 mb-4">styles appplied with tw</div>
    <div className="bg-electric text-white p-4 mb-4">styles appplied with className (no worky)</div>
    <div className={s.test}>styles appplied with postcss</div>
```

why am I asking? i'm tempted to introduce twin.macro into an existing project, however i can't rewrite all the classNames to tw now.

**Reply by ben-rogerson** [reply]
Twin can pluck tailwind classes from the className prop but you'll need to enable it.

In your twin config:

```js
// babel-plugin-macros.config.js
module.exports = {
  twin: {
    includeClassNames: true,
  },
}

```

From [the docs](https://github.com/ben-rogerson/twin.macro/blob/master/docs/options.md):

> When a tailwind class is found in a className prop, it’s plucked out, converted and delivered to the css-in-js library.
> 
> - Unmatched classes are skipped and preserved within the className
> - Suggestions aren’t shown for unmatched classes like they are for the tw prop
> - The tw and css props can be used on the same jsx element
> - Limitation: classNames with conditional props or variables aren’t touched, eg: <div className={isBlock && "block"} />

**Reply by denu5** [reply]
awesome works, thanks a lot! - shame on me for not reading the config options :)
